draft-ietf-quic-transport.md

```diff
@@ -1475,37 +1479,271 @@ failure.  If integrity protection is performed by QUIC, QUIC MUST abort the
+finished and the client has 1-RTT keys.
+
+
+### Responding to Connection Migration
+
+A server may receive a packet from a new client address at any time during the
+connection after the handshake is complete. If the packet is authenticated, the
+client might be either probing from a new address or migrating immediately, as
+described in {{initiating-migration}}.
+
+#### Responding to a Client Probe
 
-QUIC connections are identified by their 64-bit Connection ID.  QUIC's
-consistent connection ID allows connections to survive changes to the client's
-IP and/or port, such as those caused by client or server migrating to a new
-network.  Connection migration allows a client to retain any shared state with a
-connection when they move networks.  This includes state that can be hard to
-recover such as outstanding requests, which might otherwise be lost with no easy
-way to retry them.
+Receiving a PATH_CHALLENGE frame from the client indicates that the client is
+probing for server reachability on this new path.  The server sends a
+PATH_RESPONSE to the new client address as per {{migrate-validate}}, but the
+server MUST continue using the current client address for sending all other
+packets.
 
-An endpoint that receives packets that contain a source IP address and port that
-has not yet been used can start sending new packets with those as a destination
-IP address and port.  Packets exchanged between endpoints can then follow the
-new path.
+PATH_CHALLENGE, PATH_RESPONSE, and PADDING frames are "probing" frames, and
+receiving only these frames from a new client address MUST NOT result in the
+server sending any other frames to that address.
 
-Due to variations in path latency or packet reordering, packets from different
-source addresses might be reordered.  The packet with the highest packet number
-MUST be used to determine which path to use.  Endpoints also need to be prepared
-to receive packets from an older source address.
+#### Responding to an Immediate Migration
 
-An endpoint MUST validate that its peer can receive packets at the new address
-before sending any significant quantity of data to that address, or it risks
-being used for denial of service.  See {{migrate-validate}} for details.
+Receiving a packet from a new client address indicates that the client may be
+immediately migrating if
+
+* the packet is a "non-probing" packet. That is, it contains a frame other than
+PATH_CHALLENGE, PATH_RESPONSE, or PADDING; and
+
+* the packet carries a packet number that is the largest seen thus far.
+
+Upon receiving such a packet, the server MUST abandon any path validation it is
+performing with other addresses on the expectation that those validations are
+likely to fail.  Abandoning path validation primarily means ceasing subsequent
+transmissions of the PATH_CHALLENGE frame.  An endpoint MUST ignore any
+subsequently received PATH_RESPONSE frames from abandoned addresses.
```

Comment:
This could mean the client ends up spending some additional time if it wants to re-establish reachability on the other addresses / maintain that ahead of time (like is best for performance). Probably okay, but worth considering.

Reply:
Ah, I see below that this allows us to avoid issues where someone tries to migrate us and we grab it back. Looks fine in that case.